### PR TITLE
fix #8777 add missing http methods in httpClient

### LINKED ISSUE #8777: please implement http put and delete in httpClient
- Implement HTTP `PUT` and `DELETE` methods on `httpClient` and `asyncHttpClient`.

### PR DESCRIPTION
**Pull request by tonygparsehub**
Implement fix as suggested by https://github.com/nim-lang/Nim/issues/8777#issuecomment-433653022

TODO: 
* violates DRY
* no tests: i checked tests/stdlib/httpclient and there don't seem to be any existing tests for different methods. putContent and deleteContent do seem to work with httpbin though.
* documentation not updated
* other methods: HEAD, PATCH, etc

Let me know if any of these are deabreakers and I can fix them

### CODE REVIEW

**Comment by dom96** [comment]
Awesome! Nice to see you giving this a go so quickly after me writing an explanation of how to do it :)

Yeah, the code is a little too repetitive. If you can figure out how to get rid of that duplication without too much disruption to the rest of the code that would be brilliant.

**Comment by tonygparsehub** [comment]
Ok, I updated. Line 1213may cause some back compat issues for those who implemented their own PUT methods, but I think the treatment of 302 should be consistent for all methods in any case.

File: lib/pure/httpclient.nim [hunk 1]
```diff
@@ -1210,7 +1210,8 @@ proc request*(client: HttpClient | AsyncHttpClient, url: string,
   for i in 1..client.maxRedirects:
     if result.status.redirection():
       let redirectTo = getNewLocation(lastURL, result.headers)
-      result = await client.requestAux(redirectTo, httpMethod, body, headers)
+      var meth = if result.status != "307": "GET" else: httpMethod
```
Comment (Araq) [inline lib/pure/httpclient.nim]: This requires a comment. Why does status 307 enforce a "GET"?

**Comment by Araq** [comment]
Please address my concern and rebase the test case failures are my fault.

File: lib/pure/httpclient.nim (hunk 1 above)
Comment (tonygparsehub) [reply lib/pure/httpclient.nim]: It's actually the opposite. See: https://en.wikipedia.org/wiki/List_of_HTTP_status_codes#3xx_Redirection

I reversed the order of the check to be more clear, but I'm just repeating the behaviour of the previous implementation of POST

**Comment by Araq** [comment]
IMHO a link to https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/307 should be added in a comment to this code, but that's just me.

**Comment by tonygparsehub** [comment]
sure, fixed

**Comment by dom96** [comment]
Looking good! Sorry this took so long.

**Pull request merged**